src/descriptor/mod.rs

```diff
@@ -826,19 +826,22 @@ impl<Pk: MiniscriptKey> fmt::Debug for Descriptor<Pk> {
 
 impl<Pk: MiniscriptKey> fmt::Display for Descriptor<Pk> {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        match *self {
-            Descriptor::Bare(ref sub) => write!(f, "{}", sub),
-            Descriptor::Pk(ref p) => write!(f, "pk({})", p),
-            Descriptor::Pkh(ref p) => write!(f, "pkh({})", p),
-            Descriptor::Wpkh(ref p) => write!(f, "wpkh({})", p),
-            Descriptor::ShWpkh(ref p) => write!(f, "sh(wpkh({}))", p),
-            Descriptor::Sh(ref sub) => write!(f, "sh({})", sub),
-            Descriptor::Wsh(ref sub) => write!(f, "wsh({})", sub),
-            Descriptor::ShWsh(ref sub) => write!(f, "sh(wsh({}))", sub),
-            Descriptor::ShSortedMulti(ref smv) => write!(f, "sh({})", smv),
-            Descriptor::WshSortedMulti(ref smv) => write!(f, "wsh({})", smv),
-            Descriptor::ShWshSortedMulti(ref smv) => write!(f, "sh(wsh({}))", smv),
-        }
+        let desc = match *self {
+            Descriptor::Bare(ref sub) => format!("{}", sub),
+            Descriptor::Pk(ref p) => format!("pk({})", p),
+            Descriptor::Pkh(ref p) => format!("pkh({})", p),
+            Descriptor::Wpkh(ref p) => format!("wpkh({})", p),
+            Descriptor::ShWpkh(ref p) => format!("sh(wpkh({}))", p),
+            Descriptor::Sh(ref sub) => format!("sh({})", sub),
+            Descriptor::Wsh(ref sub) => format!("wsh({})", sub),
+            Descriptor::ShWsh(ref sub) => format!("sh(wsh({}))", sub),
+            Descriptor::ShSortedMulti(ref smv) => format!("sh({})", smv),
+            Descriptor::WshSortedMulti(ref smv) => format!("wsh({})", smv),
+            Descriptor::ShWshSortedMulti(ref smv) => format!("sh(wsh({}))", smv),
+        };
+        let checksum = desc_checksum(&desc).map_err(|_| fmt::Error)?;
```

Comment:
can be ignored. As corrected by Andrew on IRC, if something is already returning an Err, we can be safe and return the Err instead of a possibly risky unwrap